Sources/XcodeProj/Scheme/XCScheme+TestableReference.swift

```diff
@@ -6,7 +6,7 @@ public extension XCScheme {
         // MARK: - Attributes
 
         public var skipped: Bool
-        public var parallelizable: Bool
+        public var parallelizable: Parallelizable
```

Comment:
This may constitute as breaking change :/ 

Options:

- Introduce a new type `parallelizableValue: Parallelizable` (or some better name 😅) and keep `parallelizable: Bool` as a Bool + mark deprecated - it can translate the to  and from underlying enum property 
- Ensure a major version is released of XcodeProj

Interesting ideas but may not be sufficient:

- Leverage `ExpressibleByBooleanLiteral` https://developer.apple.com/documentation/swift/expressiblebybooleanliteral - that may mitigate any literal usage but will still break any clients using a boolean variable or reading this boolean variable :/

Reply:
I think it's worth investigating using the `ExpressibleByBooleanLiteral` to make the migration easier. Technically speaking, it won't be sufficient, so let's make a new major version of `XcodeProj`. I think for this library, having major versions more often than let's say for the Tuist CLI is fine.

Reply:
+1 this gets it a bit closer to the nomenclature in the Xcode UI

![Screenshot 2024-11-28 at 08 09 46](https://github.com/user-attachments/assets/a94385b1-8340-4d93-923c-dba4c768a6cd)

Reply:
Hey @kelvinharron - what do you think about this ? 
If you want an extra hand I'm happy to finalize this PR so we can land it into XcodeProj. LMK! 
Thanks :)

Reply:
Hey folks, I can apply these changes shortly. We're saying that we'll include the boolean but deprecate the field, would we still intend to keep the deprecated init as well?

Reply:
> Hey folks, I can apply these changes shortly. We're saying that we'll include the boolean but deprecate the field

Thanks @kelvinharron 

> Would we still intend to keep the deprecated init as well?

Ya we'll need to avoid the breaking change seeing the purpose of those additional changes is backwards compatibility

Reply:
Thanks for the feedback @kwridan, keep me right on the most recent changes as it feels a little loose with two properties. This PR extends more so than replaces now, so we're handling both mapping functions from attributes and inits as before, with the introduction of TestParallelization option.

Reply:
Ace @freak4pc. I've updated and added some tests to reason about the computed var behaviour. Please take a look and let me know if that's expected.

Reply:
@kwridan @fortmarek let me know if you have any additional feedback and I can close this out. I think we're encountering some failures on CI with these changes I can look at.